src/libraries/Microsoft.Extensions.DependencyInjection.Specification.Tests/src/ActivatorUtilitiesTests.cs

```diff
@@ -247,12 +261,23 @@ public void TypeActivatorUsesMarkedConstructor(CreateInstanceFunc createFunc)
             Assert.Equal("IFakeService, string", instance.CtorUsed);
         }
 
+        [Fact]
+        public void TypeActivatorThrowsOnNullProvider()
+        {
+            var exception = Assert.Throws<ArgumentNullException>(() => CreateInstance<ClassWithMultipleMarkedCtors>(CreateInstanceDirectly, null, "hello"));
```

Comment:
This test shouldn't be in the Specification tests, but instead in the DI.Tests\ActivatorUtilitiesTests.cs.

The Specification tests are to ensure the external DI implementations meet the minimum expected behavior. Things under `DI.Tests` ensure our behavior is correct.